faces-tck/pom.xml

```diff
@@ -0,0 +1,162 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright (c) 2021, 2024 Contributors to the Eclipse Foundation. All rights reserved.
+    Portions Copyright 2024 Payara Foundation and/or its affiliates
+
+    This program and the accompanying materials are made available under the
+    terms of the Eclipse Public License v. 2.0, which is available at
+    http://www.eclipse.org/legal/epl-2.0.
+
+    This Source Code may also be made available under the following Secondary
+    Licenses when the conditions for such availability set forth in the
+    Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+    version 2 with the GNU Classpath Exception, which is available at
+    https://www.gnu.org/software/classpath/license.html.
+
+    SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+
+-->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>fish.payara.jakarta.tests.tck</groupId>
+        <artifactId>tck</artifactId>
+        <version>1.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>payara-external-tck-faces</artifactId>
+    <name>TCK: Faces</name>
+
+    <description>This verifies the compliance of Payara using the Jakarta Faces standalone TCK</description>
+
+    <properties>
+        <faces.version>4.1.1</faces.version>
+        <tck.root>${project.build.directory}</tck.root>
+        <mojarra.version>4.1.2</mojarra.version>
+        <maven.executable>${maven.home}/bin/mvn</maven.executable>
+        <maven.settings.xml>${user.home}/.m2/settings.xml</maven.settings.xml>
+        <maven.exec.params>-B -e -s ${maven.settings.xml} -f ${tck.root}/faces-tck-${faces.version}/tck/pom.xml</maven.exec.params>
+        <maven.exec.projects></maven.exec.projects>
+        <maven.exec.goals>clean install</maven.exec.goals>
+        <maven.exec.basic.options>-Ppayara-ci-remote -P!glassfish-ci-managed -Dpayara.version=${payara.version} -Dmojarra.version=${mojarra.version} -Dfaces.version=${faces.version} -Dglassfish.version=8.0.0  -Dsigtest.api.groupId=org.glassfish -Dsigtest.api.artifactId=jakarta.faces -Dsigtest.api.version=4.1.1 -Dmojarra.noupdate</maven.exec.basic.options>
```

Comment:
I don't think we need to provide mojarra version either? mojarra.noupdate is only used in glassfish-ci-managed profile.

```suggestion
        <maven.exec.basic.options>-Ppayara-ci-remote -P!glassfish-ci-managed -Dpayara.version=${payara.version} -Dfaces.version=${faces.version} -Dsigtest.api.groupId=org.glassfish -Dsigtest.api.artifactId=jakarta.faces -Dsigtest.api.version=4.1.1</maven.exec.basic.options>
```

Reply:
It would be better if we were defining the sigtest.api.version in another way, such as by somehow pulling it from the Payara-BOM. Else we run the risk of not actually testing the artifact being bundled in Payara unless we constantly update this runner to ensure everything is aligned (which we will obviously forget to do).

Alternatively, doing some jiggery pokery - install it locally with a nonsense version:
* Mandate payara.home be provided
* Use maven-install-plugin to install the jakarta.faces.jar module bundled in Payara Server (under payara7/glassfish/modules) with a nonsense GAV e.g. fish.payara.tck.sigtest:jakarta.faces:${payara.version}
* Pass these to the TCK e.g. `-Dsigtest.api.groupId=fish.payara.tck.sigtest -Dsigtest.api.artifactId=jakarta.faces -Dsigtest.api.version=${payara.version}`

Reply:
> Else we run the risk of not actually testing the artifact being bundled in Payara unless we constantly update this runner to ensure everything is aligned (which we will obviously forget to do).

Looking at the sigtest runner, there's quite a few APIs that it uses which we may also want to look at ensuring align with what's included in Payara (e.g. jakarta.el-api, cdi-api, servlet-api etc.)

Reply:
ad payara-client-ee9: yes, we can make PR for Faces TCK. I created FISH-10173 for it.

Reply:
I upgraded payara client to ee11, nothing changed.